Cesium.CodeGen/Ir/TopLevel/FunctionDefinition.cs

```diff
@@ -122,22 +129,15 @@ private bool ValidateMainParameters()
                 $"Invalid parameter types for the {_name} function: " +
                 "int, char*[] expected.");
 
-        return true;
+        return GenerateSyntheticEntryPointStrArray(context, userEntrypoint);
     }
 
-    /// <summary>
-    /// One of the standard signatures for the main function is <code>int main(int argc, char *argv[])</code>. This
-    /// isn't directly supported by the CLI infrastructure, so we'll have to emit a synthetic entrypoint for this case,
-    /// which will accept a managed string array and prepare the arguments for the C function.
-    /// </summary>
-    /// <returns>A synthetic entrypoint method created.</returns>
-    private MethodDefinition GenerateSyntheticEntryPoint(
+    private MethodDefinition GenerateSyntheticEntryPointStrArray(
         TranslationUnitContext context,
         MethodReference userEntrypoint)
     {
-        var module = context.Module;
         var syntheticEntrypoint = new MethodDefinition(
-            "<SyntheticEntrypoint>",
+            "<SyntheticEntrypoint>StrArray",
```

Comment:
_(Minor)_ Do we have to use different function names for different `main` variants? I don't think so, but I'm ready to listen to the arguments.

Reply:
Technically I suspect that MSVC team do this for some reasons, but let's keep all syntetic entry points as close as possible. There even slightest idea right now in my mind.

Reply:
Well, that would be an argument, but whatever: you've already changed the name back. Okay!

Reply:
We don’t know and better not have hint to ancient knowledge like this. Most likely we forget about it. And if that’s really needed we definitely hit some issue and can make change

Reply:
Agree. Let the ancient ones keep their secrets.